### PR TITLE
Sent suggest audit events

### PR DESCRIPTION
**Pull request by sunbit**
Include client type info in search and suggest events Add the jaeger trace id to audit

### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb/nucliadb/search/api/v1/suggest.py [hunk 1]
```diff
@@ -179,4 +183,14 @@ async def suggest_knowledgebox(
...
     if debug:
         search_results.shards = queried_shards
+
+    if audit is not None:
+        await audit.suggest(
+            kbid,
+            x_nucliadb_user,
+            x_ndb_client.to_proto(),
+            x_forwarded_for,
+            timeit - time(),
```
Comment (lferran) [inline nucliadb/nucliadb/search/api/v1/suggest.py:193]: this should be
```python
time() - timeit
```

File: nucliadb/nucliadb/tests/test_search.py [hunk 2]
```diff
@@ -952,3 +958,76 @@ async def test_only_search_calls_audit(nucliadb_reader, knowledgebox):
...
+
+    # Test suggest sends audit
+    resp = await nucliadb_reader.get(
+        f"/kb/{kbid}/suggest?query=", headers={"x-ndb-client": "chrome_extension"}
+    )
+    assert resp.status_code == 200
+
+    auditreq = await get_audit_messages(psub)
+
+    assert auditreq.kbid == kbid
```
Comment (lferran) [inline nucliadb/nucliadb/tests/test_search.py:1027]: To be more confident, you could have assertions for all attributes of `auditreq`

File: nucliadb_models/nucliadb_models/search.py [hunk 3]
```diff
@@ -31,10 +32,10 @@
...
 from nucliadb_models.resource import ExtractedDataTypeName, Resource
 
-if TYPE_CHECKING:
-    SortValue = OrderBy.OrderType.V
-else:
-    SortValue = int
+# if TYPE_CHECKING:
+#     ClientTypeValue = ClientType.V
+# else:
+#     ClientTypeValue = int
```
Comment (lferran) [inline nucliadb_models/nucliadb_models/search.py:38]: does this need to be uncommented back?

File: nucliadb_utils/nucliadb_utils/audit/basic.py [hunk 4]
```diff
@@ -41,12 +41,23 @@ async def search(
...
     ):
         logger.debug(f"SEARCH {kbid} {user} {origin} ''{search}'' {timeit} {resources}")
 
+    async def suggest(
+        self,
+        kbid: str,
+        user: str,
+        client: int,
+        origin: str,
+        timeit: float,
```
Comment (lferran) [inline nucliadb_utils/nucliadb_utils/audit/basic.py:58]: what about `duration` as a better name?

File: nucliadb_utils/nucliadb_utils/audit/stream.py [hunk 5]
```diff
@@ -175,12 +180,34 @@ async def search(
...
+
+    async def suggest(
+        self,
+        kbid: str,
+        user: str,
+        client_type: int,
+        origin: str,
+        timeit: float,
+    ):
+        # Search is a base64 encoded search
```
Comment (lferran) [inline nucliadb_utils/nucliadb_utils/audit/stream.py:202]: this comment does not belong here right?

**Review by lferran — CHANGES_REQUESTED** [review]

File: nucliadb/nucliadb/search/api/v1/suggest.py (hunk 1 above)
Comment (sunbit) [reply nucliadb/nucliadb/search/api/v1/suggest.py:193]: wow good catch. There's at least one other place where this is wrong, not from this PR; i'll fix it

File: nucliadb_models/nucliadb_models/search.py (hunk 3 above)
Comment (sunbit) [reply nucliadb_models/nucliadb_models/search.py:38]: i'll remove it,  i could't find a way to make that work with enums, it was just breaking in one point of another, so i typ ignored the two offending lines in the stream.py file

File: nucliadb_utils/nucliadb_utils/audit/basic.py (hunk 4 above)
Comment (sunbit) [reply nucliadb_utils/nucliadb_utils/audit/basic.py:58]: Agree, let me check which consequences may have this change on the receiving side of this events

File: nucliadb_utils/nucliadb_utils/audit/stream.py (hunk 5 above)
Comment (sunbit) [reply nucliadb_utils/nucliadb_utils/audit/stream.py:202]: yup, copy paste

File: nucliadb/nucliadb/search/api/v1/suggest.py (hunk 1 above)
Comment (sunbit) [reply nucliadb/nucliadb/search/api/v1/suggest.py:193]: Fixed

File: nucliadb_models/nucliadb_models/search.py (hunk 3 above)
Comment (sunbit) [reply nucliadb_models/nucliadb_models/search.py:38]: Fixed

File: nucliadb_utils/nucliadb_utils/audit/stream.py (hunk 5 above)
Comment (sunbit) [reply nucliadb_utils/nucliadb_utils/audit/stream.py:202]: Fixed

File: nucliadb_utils/nucliadb_utils/audit/basic.py (hunk 4 above)
Comment (sunbit) [reply nucliadb_utils/nucliadb_utils/audit/basic.py:58]: We have some places that make use of this field, so i'll leave it like this for now.

File: nucliadb/nucliadb/tests/test_search.py (hunk 2 above)
Comment (sunbit) [reply nucliadb/nucliadb/tests/test_search.py:1027]: from the POV of search, this are enough IMO, I agree that we may have unit tests on the `nucliadb_utils` for this that we don't right now, and we really need this to get rolling.

**Review by lferran — APPROVED** [review]

**Pull request merged**